static/panes/output.ts

```diff
@@ -0,0 +1,273 @@
+            ...parent,
+        };
+        this.fontScale.addState(state);
+        return state as any;
+    }
+
+    addOutputLines(result) {
+        for(const obj of (result.stdout || []).concat(result.stderr || [])) {
+            const lineNumber = obj.tag ? obj.tag.line : obj.line;
+            const columnNumber = obj.tag ? obj.tag.column : -1;
+            if (obj.text === '') {
+                this.add('<br/>');
+            } else {
+                this.add(this.normalAnsiToHtml.toHtml(obj.text),
+                    lineNumber, columnNumber, obj.tag ? obj.tag.file : false);
+            }
+        }
+    }
+
+    onCompiling(compilerId: number) {
+        if (this.compilerInfo.compilerId === compilerId) {
+            this.setCompileStatus(true);
+        }
+    }
+
+    override onCompileResult(compilerId: number, compiler: any, result: any) {
+        if (compilerId !== this.compilerInfo.compilerId) return;
+        if (compiler) this.compilerInfo.compilerName = compiler.name;
+
+        this.contentRoot.empty();
+
+        if (result.buildsteps) {
+            for(const step of result.buildsteps) {
+                this.add('Step ' + step.step + ' returned: ' + step.code);
+                this.addOutputLines(step);
+            }
+        } else {
+            this.addOutputLines(result);
+            if (!result.execResult) {
+                this.add('Compiler returned: ' + result.code);
+            } else {
+                this.add('ASM generation compiler returned: ' + result.code);
+                this.addOutputLines(result.execResult.buildResult);
+                this.add('Execution build compiler returned: ' + result.execResult.buildResult.code);
+            }
+        }
+
+        if (result.execResult && (result.execResult.didExecute || result.didExecute)) {
+            this.add('Program returned: ' + result.execResult.code);
+            if (result.execResult.stderr.length || result.execResult.stdout.length) {
+                for(const obj of result.execResult.stderr) {
+                    // Conserve empty lines as they are discarded by ansiToHtml
+                    if (obj.text === '') {
+                        this.programOutput('<br/>');
+                    } else {
+                        this.programOutput(this.errorAnsiToHtml.toHtml(obj.text), 'red');
+                    }
+                }
+
+                for(const obj of result.execResult.stdout) {
+                    // Conserve empty lines as they are discarded by ansiToHtml
+                    if (obj.text === '') {
+                        this.programOutput('<br/>');
+                    } else {
+                        this.programOutput(this.normalAnsiToHtml.toHtml(obj.text));
+                    }
+                }
+            }
+        }
+        this.setCompileStatus(false);
+        this.updateTitle();
+    }
+
+    override onCompiler(compilerId: number, compiler: unknown, options: unknown, editorId: number,
```

Comment:
where is this coming from exactly? i dont see it in the original .js file

Reply:
`onCompiler` is abstract in pane.ts so we have to provide an implementation even if it the pane doesn't do anything for the event